src/controls/UserEditInteraction.js

```diff
@@ -0,0 +1,218 @@
+    this.#map = options.map;
+
+    this.#editLayer = new VectorLayer({
+      source: new VectorSource({
+        features: [],
+      }),
+    });
+
+    this.drawInteraction = new Draw({
+      source: this.#editLayer.getSource(),
+      condition: event => {
+        const existingFeature = event.map
+          .getFeaturesAtPixel(event.pixel, {
+            hitTolerance: 7,
+          })
+          .find(f => {
+            return f.get('layer') === this.#yioMap.editLayer;
+          });
+        return !existingFeature;
+      },
+      type: 'Point',
+    });
+    this.drawInteraction.on('drawend', event => {
+      const feature = event.feature;
+      feature.set('layer', this.#yioMap.editLayer);
+      this.#editLayer.getSource().once('addfeature', () => {
+        this.#handleClick(event);
+      });
+    });
+    this.modifyInteraction = new Modify({
+      source: this.#editLayer.getSource(),
+    });
+
+    this.modifyInteraction.on('modifystart', event => {
+      const modifiedFeatureIds = event.features
+        .getArray()
+        .map(feature => getFeatureId(feature))
+        .filter(id => !!id);
+      if (modifiedFeatureIds.length) {
+        this.#modifiedFeatureIDs.push(...modifiedFeatureIds);
+        this.#olSourceLayer.changed();
+      }
+    });
+    this.modifyInteraction.on('modifyend', event => {
+      this.#handleClick(event);
+    });
+    this.drawInteraction.setActive(false);
+    this.modifyInteraction.setActive(false);
+  }
+
+  setMap(map) {
+    const currentMap = this.getMap();
+    if (currentMap) {
+      currentMap.removeLayer(this.#editLayer);
+    }
+    if (!map) {
+      return;
+    }
+    map.addLayer(this.#editLayer);
+    super.setMap(map);
+    this.modifyInteraction.setMap(this.#map);
+    this.drawInteraction.setMap(this.#map);
+  }
+
+  drawInteraction;
+
+  modifyInteraction;
+
+  /**
+   * @param {import('ol/MapBrowserEvent.js').default} event
+   */
+  #handleClick(event) {
+    this.#yioMap.dispatchEvent(new PointerEvent('click', event.originalEvent));
+  }
+
+  /**
+   * original style of the content layer
+   */
+  #originalStyle = null;
+
+  setActive(active) {
+    if (!this.modifyInteraction || !this.drawInteraction) {
+      return;
+    }
+    const sourceLayer = getOlLayerForMbLayer(
+      this.#yioMap._getContentLayer(),
+      this.#yioMap.editLayer,
+    );
+    if (active) {
+      this.#olSourceLayer = sourceLayer;
+      this.#originalStyle = sourceLayer.getStyle();
+      sourceLayer.setStyle((feature, resolution) => {
+        if (this.#modifiedFeatureIDs.includes(getFeatureId(feature))) {
+          return;
+        }
+        return this.#originalStyle(feature, resolution);
+      });
+    } else {
+      if (this.#originalStyle) {
+        this.#olSourceLayer.setStyle(this.#originalStyle);
+      }
+      this.#olSourceLayer = null;
+    }
+    this.modifyInteraction.setActive(!!active);
+    this.drawInteraction.setActive(!!active);
```

Comment:
Better to give the `active` argument a `boolean` type in the function signature's JSDoc than adding code for just satisfying the type:

```suggestion
    this.modifyInteraction.setActive(active);
    this.drawInteraction.setActive(active);
```